Pa3/Durchführung/Project/app/State/CalibrateLineSensors.c

```diff
@@ -9,24 +9,201 @@
 /**************************************************************************************************/
 
 /* INCLUDES ***************************************************************************************/
-#include "CalibrateLineSensors.h"
+#include "app/State/CalibrateLineSensors.h"
+
+#include "service/DriveControl.h"
+#include "service/LineSensor.h"
+
+#include "os/Task.h"
+#include "os/Scheduler.h"
 
-#include "DriveControl.h"
-#include "LineSensor.h"
 /* CONSTANTS **************************************************************************************/
+#define CALIB_SPEED 33u       /**< Motor speed while calibrating */
+#define CALIB_SPEED_SLOW 25u  /**< Motor speed while centering on line  */
 
 /* MACROS *****************************************************************************************/
 
 /* TYPES ******************************************************************************************/
+/** Calibration state. */
+typedef enum tag_CalibrationState
+{
+    CALIBRATION_STATE_INIT,                             /**< Initial calibration state.           */
+    CALIBRATION_STATE_TURN_RIGHT_UNTIL_LEFT_SENSOR,     /**< State TURN_RIGHT_UNTIL_LEFT_SENSOR.  */
+    CALIBRATION_STATE_TURN_LEFT_UNTIL_RIGHT_SENSOR,     /**< State TURN_LEFT_UNTIL_RIGHT_SENSOR.  */
+    CALIBRATION_STATE_CENTER_ON_LINE,                   /**< State CENTER_ON_LINE.                */
+    CALIBRATION_STATE_FINISHED,                         /**< State FINISHED.                      */
+    CALIBRATION_STATE_TIMEOUT,                          /**< State TIMEOUT.                       */
+    CALIBRATION_STATE_DONE                              /**< State DONE.                          */
+} CalibrationState;
 
 /* PROTOTYPES *************************************************************************************/
 
+/** Reaction to processCycle event.
+ * 
+ * Handles internal state machine processing.
+ *
+ * @param[in] pState Pointer to context state.
+ */
+static void onProcessCycleEvent (StateHandlerStatePtr pState);
+
 /* VARIABLES **************************************************************************************/
 
+/** Calibration state of local state machine */
+static CalibrationState gState;
+
+/** Calibration result of local state machine */
+static EventEnum gResult;
+
+/** Timer used by calibration steps. */
+static SoftTimer gTimer;
+
+/** CalibrationTask task structure. */
+static Task gCalibrationTask;
+
 /* EXTERNAL FUNCTIONS *****************************************************************************/
+
+EventEnum CalibrateLineSensors_Initialize(void)
+{
+    EventEnum result = CALIBRATION_FAILED;
+    gState = CALIBRATION_STATE_INIT;
+    gResult = NO_EVENT_HAS_HAPPEND;
+
+    //Add Task to scheduler
```

Comment:
see CodingStyleGuide for in-function documentation

```suggestion
    /* Add task to scheduler. */
```